packages/fast-stable-stringify/src/__tests__/index-test.ts

```diff
@@ -0,0 +1,201 @@
+                5e-324,
+                -1.7976931348623157e308,
+                -5e-324,
+                Infinity,
+                -Infinity,
+                NaN,
+            ],
+        ],
+        [
+            'arrays of strings',
+            [
+                'a b c',
+                'abc',
+                'ABC',
+                'NUMBER_ONLY',
+                '',
+                '\u0000\u001F',
+                '\u0020\uFFFF',
+                '☃',
+                '"',
+                '\\',
+                '/',
+                '\f',
+                '\n',
+                '\r',
+                '\t',
+                '\b',
+                'Aa1 Bb2 Cc3 \u0000\u001F\u0020\uFFFF☃"\\/\f\n\r\t\b',
+            ],
+        ],
+        ['arrays of booleans ', [true, false]],
+        ['arrays of null', [null]],
+        ['arrays of undefined', [undefined]],
+        ['arrays of Date instances', [new Date('2017')]],
+        ['arrays of instances that implement `toJSON`', [new ImplementingToJSON()]],
+        ['arrays of instances that do not implement `toJSON`', [new NotImplementingToJSON()]],
+        ['arrays of functions', [function () {}]],
+        [
+            'arrays of mixed values',
+            [
+                -1.7976931348623157e308,
+                -5e-324,
+                'Aa1 Bb2 Cc3 \u0000\u001F\u0020\uFFFF☃"\\/\f\n\r\t\b',
+                true,
+                false,
+                null,
+                undefined,
+            ],
+        ],
+        [
+            'mixed values',
+            [
+                {
+                    'Aa1 Bb2 Cc3 \u0000\u001F\u0020\uFFFF☃"\\/\f\n\r\t\b': 'MIXED',
+                    DATE: new Date('2017'),
+                    FALSE: false,
+                    FUNCTION: function () {},
+                    IMPLEMENTING_TO_JSON: new ImplementingToJSON(),
+                    MAX_VALUE: 1.7976931348623157e308,
+                    MIN_VALUE: 5e-324,
+                    MIXED: 'Aa1 Bb2 Cc3 \u0000\u001F\u0020\uFFFF☃"\\/\f\n\r\t\b',
+                    NEGATIVE_MAX_VALUE: -1.7976931348623157e308,
+                    NEGATIVE_MIN_VALUE: -5e-324,
+                    NOT_IMPLEMENTING_TO_JSON: new NotImplementingToJSON(),
+                    NULL: null,
+                    TRUE: true,
+                    UNDEFINED: undefined,
+                    zzz: 'ending',
+                },
+                -1.7976931348623157e308,
+                -5e-324,
+                'Aa1 Bb2 Cc3 \u0000\u001F\u0020\uFFFF☃"\\/\f\n\r\t\b',
+                true,
+                false,
+                null,
+                undefined,
+                new Date('2017'),
+                function () {},
+                new ImplementingToJSON(),
+                new NotImplementingToJSON(),
+            ],
+        ],
+    ])('matches the output of `json-stable-stringify` when hashing %s (`%s`)', (_, value) => {
+        expect(stringify(value)).toBe(jsonStableStringify(value));
```

Comment:
Do you want to bench this against anything else, like `faster-stable-stringify` or `JSON.stringify`?

Reply:
The original `fast-stable-stringify` tests verified its output against `json-stable-stringify` for correctness only.

I at least wanted to use that as a starting point so that as we make changes we can observe how they make ours diverge.